java/src/main/java/ai/rapids/cudf/DType.java

```diff
@@ -116,88 +268,111 @@ public boolean isBackedByInt() {
    *       DType.TIMESTAMP_NANOSECONDS
    */
   public boolean isBackedByLong() {
-    return LONGS.contains(this);
+    return LONGS.contains(this.typeId);
   }
 
+  /**
+   * Returns true if this type is backed by short type
+   * Namely this method will return true for the following types
+   *       DType.INT16,
+   *       DType.UINT16
+   */
+  public boolean isBackedByShort() { return SHORTS.contains(this.typeId); }
+
+  /**
+   * Returns true if this type is backed by byte type
+   * Namely this method will return true for the following types
+   *       DType.INT8,
+   *       DType.UINT8,
+   *       DType.BOOL8
+   */
+  public boolean isBackedByByte() { return BYTES.contains(this.typeId); }
+
+  /**
+   * Returns true if this type is backed by decimal type
+   * Namely this method will return true for the following types
+   *       DType.DECIMAL32,
+   *       DType.DECIMAL64
+   */
+  public boolean isBackedByDecimal() { return DECIMALS.contains(this.typeId); }
```

Comment:
to me `isBackedBy` indicates how the data is stored. I would rather see this be called `isDecimalType` to match `isDurationType` and `isNestedType`.